validations/src/validations.rs

```diff
@@ -1089,6 +1116,59 @@ pub fn validate_tally_transaction<'a>(
     Ok((ta_tx.outputs.iter().collect(), tally_extra_fee))
 }
 
+/// Function to validate a stake transaction.
+pub fn validate_stake_transaction<'a>(
+    st_tx: &'a StakeTransaction,
+    utxo_diff: &UtxoDiff<'_>,
+    epoch: Epoch,
+    epoch_constants: EpochConstants,
+    signatures_to_verify: &mut Vec<SignaturesToVerify>,
+) -> Result<
+    (
+        Vec<&'a Input>,
+        &'a StakeOutput,
+        u64,
+        u32,
+        &'a Option<ValueTransferOutput>,
+    ),
+    failure::Error,
+> {
+    // Check that the amount of coins to stake is equal or greater than the minimum allowed
+    if st_tx.body.output.value < MIN_STAKE_NANOWITS {
```

Comment:
See my former comment on consistency of `?` vs. `return` syntax inside the same function.